packages/xo-web/src/common/xo/snapshot-vm-modal/index.js

```diff
@@ -13,29 +13,43 @@ import { buildTemplate, connectStore } from 'utils'
   { withRef: true }
 )
 export default class SnapshotVmModalBody extends BaseComponent {
-  state = { namePattern: '{name}_{date}' }
+  state = {
+    descriptionPattern: '{description}',
+    namePattern: '{name}_{date}',
+  }
 
   get value() {
-    const { namePattern, saveMemory } = this.state
-    if (namePattern === '') {
-      return { names: {}, saveMemory }
+    const { descriptionPattern, namePattern, saveMemory } = this.state
+    if (namePattern === '' && descriptionPattern === '') {
+      return { names: {}, descriptions: {}, saveMemory }
     }
 
-    const generateName = buildTemplate(namePattern, {
+    const RULES = {
+      '{date}': () => new Date().toISOString(),
+      '{description}': vm => vm.name_description,
       '{name}': vm => vm.name_label,
-      '{date}': new Date().toISOString(),
+    }
+    const generateName = buildTemplate(namePattern, RULES)
+    const generateDescription = buildTemplate(descriptionPattern, RULES)
+
+    const names = []
+    const descriptions = []
+    forEach(this.props.vms, (vm, id) => {
+      names[id] = generateName(vm)
+      descriptions[id] = generateDescription(vm)
     })
 
     return {
-      names: mapValues(this.props.vms, generateName),
+      names: namePattern !== '' ? names : {},
+      descriptions: descriptionPattern !== '' ? descriptions : {},
```

Comment:
Is this the behaviour we want? If the user explicitly removes the default text from the input and leaves it empty, doesn't it mean that he wants the snapshots' descriptions to be empty?

Reply:
The description will be the description of the VM, if the user removes the default text.